### PR TITLE
Document the RFC process better.

### PR DESCRIPTION
**Pull request by knz**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4056)

### CODE REVIEW

File: docs/RFCS/README.md [hunk 1]
```diff
@@ -9,8 +9,10 @@ Before making major changes, consider sending an RFC for discussion.
 This is a lightweight design-document process that is inspired by the
 [process used by the rust project](https://github.com/rust-lang/rfcs).
 
-1. Copy `template.md` to a new file and fill in the details.
-2. Submit a pull request to add your new file. Each RFC should get its
+1. Copy `template.md` or `template.rst` to a new file and fill in the details.
+   Commit this version in your own fork of the repository or a branch.
+2. Submit a pull request to add your new file to the main repository.
+   Each RFC should get its
```
Comment (tschottdorf) [inline docs/RFCS/README.md]: nit: unnatural linebreak

File: docs/RFCS/README.md [hunk 2]
```diff
@@ -21,3 +23,33 @@ This is a lightweight design-document process that is inspired by the
...
+
+Draft
+   A newly minted RFC has this status, until either
+   the proposal is accepted (next status:
+   in-progress) or that it is DOA (next status: rejected).
+
+In-progress
+   A RFC receives this status when the PR discussions have concluded
+   the proposal should be implemented, and the RFC is ready to commit
+   (merge) in the main repository.  Next status: completed (when the
```
Comment (tschottdorf) [inline docs/RFCS/README.md]: s/in/to/

File: docs/RFCS/README.md [hunk 3]
```diff
@@ -21,3 +23,33 @@ This is a lightweight design-document process that is inspired by the
...
+   work is done), rejected (proposal not implemented/implementable
+   after all), obsolete (some subsequent work removes the need for the
+   feature).
+
+Completed
+   A RFC receives this status when the described feature has been
+   implemented. Next status: obsolete.
```
Comment (tschottdorf) [inline docs/RFCS/README.md]: I don't understand the "Next status: obsolete" part.

File: docs/RFCS/README.md [hunk 4]
```diff
@@ -21,3 +23,33 @@ This is a lightweight design-document process that is inspired by the
...
+
+When you mark a RFC as obsolete, ensure that its text references the
+other RFCs or PRs that make it obsolete.
+
+# RFC Status
+
+During its lifetime a RFCs can have the following status:
+
+Draft
+   A newly minted RFC has this status, until either
```
Comment (tschottdorf) [inline docs/RFCS/README.md]: ditto (in this paragraph)

File: docs/RFCS/README.md [hunk 5]
```diff
@@ -21,3 +23,33 @@ This is a lightweight design-document process that is inspired by the
...
    developments render an RFC obsolete, set its status to `obsolete`.
+
+When you mark a RFC as obsolete, ensure that its text references the
+other RFCs or PRs that make it obsolete.
+
+# RFC Status
+
+During its lifetime a RFCs can have the following status:
+
+Draft
```
Comment (tschottdorf) [inline docs/RFCS/README.md]: this doesn't display correctly, try something like

```
- Draft
  Lorem Ipsum
```

**Comment by tschottdorf** [comment]
LGTM

**Comment by knz** [comment]
Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[docs/RFCS/README.md, line 15 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r1-15:-K9RtgK4TX1X9vx0eZ8v:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/01d29ee471cd601b8fc9fceb9ffa69acb95262c1/docs/RFCS/README.md#L15)):*
Done.

---

*[docs/RFCS/README.md, line 34 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r1-34:-K9Rth_2HbnAmwGKaBQa:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/01d29ee471cd601b8fc9fceb9ffa69acb95262c1/docs/RFCS/README.md#L34)):*
Done.

---

*[docs/RFCS/README.md, line 35 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r1-35:-K9RtofCzMxa2ZH3FBTY:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/01d29ee471cd601b8fc9fceb9ffa69acb95262c1/docs/RFCS/README.md#L35)):*
Done.

---

*[docs/RFCS/README.md, line 42 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r1-42:-K9RtjJuAifcp0iOypWh:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/01d29ee471cd601b8fc9fceb9ffa69acb95262c1/docs/RFCS/README.md#L42)):*
Done.

---

*[docs/RFCS/README.md, line 49 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r1-49:-K9RtqvHOMfzRpTjOjdi:693357124) ([raw file](https://github.com/cockroachdb/cockroach/blob/01d29ee471cd601b8fc9fceb9ffa69acb95262c1/docs/RFCS/README.md#L49)):*
It's the next possible status, if anything ever changes afterwards

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056)*

File: docs/RFCS/README.md [hunk 6]
```diff
@@ -9,15 +9,57 @@ Before making major changes, consider sending an RFC for discussion.
...
    developments render an RFC obsolete, set its status to `obsolete`.
+
+When you mark a RFC as obsolete, ensure that its text references the
+other RFCs or PRs that make it obsolete.
+
+# RFC Status
+
+During its lifetime a RFCs can have the following status:
+
+- Draft
```
Comment (vivekmenezes) [inline docs/RFCS/README.md]: Is there value to Draft versus Accepted/Rejected?

File: docs/RFCS/README.md [hunk 7]
```diff
@@ -9,15 +9,57 @@ Before making major changes, consider sending an RFC for discussion.
...
+
+- In-progress
+  
+  A RFC receives this status when the PR discussions have concluded
+  the proposal should be implemented, and the RFC is ready to commit
+  (merge) to the main repository.
+
+  Next possible status: completed (when the work is done), rejected
+  (proposal not implemented/implementable after all), obsolete (some
+  subsequent work removes the need for the feature).
```
Comment (vivekmenezes) [inline docs/RFCS/README.md]: I think we can just call this Accepted, and draft would be an empty status field. In-progress implies work has commenced on the project which is usually better managed within the issue (usually someone assigned to an issue implies that they are working on it). Sometimes an RFC can be published and  work might not commence for a year on it.

**Comment by knz** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/RFCS/README.md, line 38 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r2-38:-K9SHNa0vy6WtKS9rHlc:-266571030) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf86ba43cbcf4503711a9b094e0d4351dc58ddeb/docs/RFCS/README.md#L38)):*
"Draft" is the status as long as the discussion is ongoing, before implementation starts (or initial implementation works confirms the idea was good).  "In-progress" indicates the discussion has completed and engineering work is ongoing. Unless you have a better suggestion?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056)*

**Comment by vivekmenezes** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/RFCS/README.md, line 38 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9RC9ylAgHv_rgdHbXX-r2-38:-K9SJUBIGtuyM8ML29Y4:1741933076) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf86ba43cbcf4503711a9b094e0d4351dc58ddeb/docs/RFCS/README.md#L38)):*
I like Draft, Accepted, Rejected, Completed as the four states. I'm not fond of in-progress

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056)*

**Comment by petermattis** [comment]
LGTM

---

Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/RFCS/README.md, line 36 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-K9SL32xKktInRDNjre5:-K9SL32xKktInRDNjre6:1421522838) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf86ba43cbcf4503711a9b094e0d4351dc58ddeb/docs/RFCS/README.md#L36)):*
`s/a/an/g`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4056#-:-K9SLnB_ulqdOYDuQsq9:676834233)*

File: docs/RFCS/README.md [hunk 8]
```diff
@@ -9,15 +9,57 @@ Before making major changes, consider sending an RFC for discussion.
 This is a lightweight design-document process that is inspired by the
 [process used by the rust project](https://github.com/rust-lang/rfcs).
 
-1. Copy `template.md` to a new file and fill in the details.
-2. Submit a pull request to add your new file. Each RFC should get its
-   own pull request; do not combine RFCs with other files.
+1. Copy `template.md` or `template.rst` to a new file and fill in the
```
Comment (bdarnell) [inline docs/RFCS/README.md]: Why do we need two different formats? Unless there is a good reason to introduce RST I'd rather standardize on markdown. Everyone's familiar with markdown through its use in github comments and other places, and markdown on github enjoys some extra features like auto-linking issue numbers.

**Pull request merged**